src/aotextarea.cpp

```diff
@@ -42,6 +42,36 @@ void AOTextArea::append_chatmessage(QString p_name, QString p_message,
   this->auto_scroll(old_cursor, old_scrollbar_value, is_scrolled_down);
 }
 
+void AOTextArea::append_ghost(QString p_name, QString p_message)
```

Comment:
This function isn't being used, which suggests that append_ic_text is doing more than it's supposed to. I'm also concerned about not saving the text cursor because there may be something that gets added later that breaks the assumption about how text blocks work - e.g. what if messages aren't 1:1 with text blocks. Also it resets the user's scroll if they are scrolling through chat.